### DISCUSSION TITLE
Gitlab for custom repo?

### DISCUSSION THREAD

**Discussion by radioraheem** [microsoft/winget-pkgs#37045] [Q&A] [answered]
Can gitlab be used to host a custom repository?  I've only been able to find one person reporting success with it but they didn't say anything about the structure/etc. 

I've created my repo on the site, created the necessary manifests using the wingetcreate tool, but I can't seem to get winget to accept it as a new source.  I get a variety of error messages depending on the target I point winget source add at.

Thanks in advance for any advice.

**Comment by florelis** [comment, marked as answer]
winget does not use git repos as sources. The manifests you see in this repo are processed into an index database, and that database is given to winget as an MSIX to be used as a source.

To host a custom repository you can implement a REST API to act as the source and point winget to it. There is a [reference implementation](https://github.com/microsoft/winget-cli-restsource) of this API that you can use, but I believe you can implement it any way you want as long as you follow the same API.
